app/controllers/sys_controller.rb

```diff
@@ -99,14 +101,41 @@ def check_enabled
   private
 
   def require_basic_auth
+    @authenticated_user = get_authenticated_user
```

Comment:
Agree with @machisuji that this is best refactored into method for sake of legibility. (see l110ff.).